packages/web-pkg/tests/unit/components/CustomComponentTarget.spec.ts

```diff
@@ -0,0 +1,156 @@
+    type: 'customComponent',
+    multiple: false
+  })
+  const mockExtensionPointMulti = mock<ExtensionPoint>({
+    id: 'dummy-extension-point-multi',
+    type: 'customComponent',
+    multiple: true
+  })
+
+  describe('no extensions match the extension point', () => {
+    it.each([mockExtensionPointSingle, mockExtensionPointMulti])(
+      'renders nothing',
+      (extensionPoint) => {
+        const { wrapper } = getWrapper({
+          extensionPoint,
+          extensions: []
+        })
+        expect(wrapper.find(selectors.target).exists()).toBeFalsy()
+      }
+    )
+  })
+
+  describe('exactly 1 extension matches the extension point', () => {
+    it.each([mockExtensionPointSingle, mockExtensionPointMulti])(
+      'renders 1 component',
+      (extensionPoint) => {
+        const extensionId = 'custom-1'
+        const { wrapper } = getWrapper({
+          extensionPoint,
+          extensions: [createExtensionMock(extensionId, extensionPoint.id)]
+        })
+        expect(wrapper.find(selectors.target).exists()).toBeTruthy()
+      }
+    )
+  })
+
+  describe('multiple extensions match the extension point', () => {
+    describe('extension point allows only 1 extension', () => {
+      it('renders 1 component, respecting the user preference', () => {
+        const extensionMocks = ['custom-1', 'custom-2'].map((id) =>
+          createExtensionMock(id, mockExtensionPointSingle.id)
+        )
+        const { wrapper } = getWrapper({
+          extensionPoint: mockExtensionPointSingle,
+          extensions: extensionMocks,
+          preference: mock<ExtensionPreferenceItem>({
+            extensionPointId: mockExtensionPointSingle.id,
+            selectedExtensionIds: [extensionMocks[1].id]
+          })
+        })
+        expect(wrapper.findAll(selectors.target).length).toBe(1)
+      })
+    })
+
+    describe('multiple extensions match the extension point', () => {
+      it('renders n components', () => {
+        const extensionMocks = ['custom-1', 'custom-2'].map((id) =>
+          createExtensionMock(id, mockExtensionPointMulti.id)
+        )
+        const { wrapper } = getWrapper({
+          extensionPoint: mockExtensionPointMulti,
+          extensions: extensionMocks
+        })
+        expect(wrapper.findAll(selectors.target).length).toBe(extensionMocks.length)
+      })
+    })
+  })
+})
+
+function getWrapper({
+  extensionPoint,
+  extensions,
+  preference
+}: {
+  extensionPoint: ExtensionPoint
+  extensions: Extension[]
+  preference?: ExtensionPreferenceItem
+}) {
+  vi.mocked(useExtensionRegistry).mockImplementation(() =>
+    useExtensionRegistryMock({
+      requestExtensions<ExtensionType>(type: string) {
+        return extensions as ExtensionType[]
+      }
+    })
+  )
+  vi.mocked(useExtensionPreferencesStore).mockImplementation(() =>
+    useExtensionPreferencesStoreMock({
+      getExtensionPreference() {
+        return preference
+      }
+    })
+  )
```

Comment:
You can mock pinia stores directly, so there is no need for a `useExtensionPreferencesStoreMock`. Example:

```
// basically moved from line 135 to here.
// the method initializes pinia, hence this needs to go before mocking it.
const plugins = defaultPlugins()

const { getExtensionPreference } = useExtensionPreferencesStore()
vi.mocked(getExtensionPreference).mockReturnValue(preference)
```

Reply:
thank you! done :-) also converted the extensionRegistry store from an optionsStore to a setupStore as discussed. Could you check my last 2 commits?